tests/test_zlib.rs

```diff
@@ -0,0 +1,331 @@
+
+    // Bitcode extracted from an executable linked against a shared library is
+    // just the contents of the executable code, but does *not* include any of
+    // the bitcode from the shared library.
+
+    let ll = get_llvm_str(&bc_path2)?;
+    let fnd = vec![
+        ll.contains("inflateCopy"),   // shared lib
+        ll.contains("compressBound"),   // shared lib
+        ll.contains("deflateSetHeader"),   // shared lib
+        ll.contains("gzfread"),   // shared lib
+        ll.contains("gzfwrite"),   // shared lib
+        ll.contains("test_gzio"),  // exe src
+    ];
+    assert_eq!(fnd, vec![false, false, false, false, false, true]);
+    assert!(tdir.path().exists()); // ensure tempdir lasts until here
+    Ok(())
+}
+
+#[test]
+#[serial]
+fn test_zlib_exe_modified() -> anyhow::Result<()> {
+    // In this test, one of the files that is part of the static library is
+    // modified, and the library is rebuilt *without* using build-bom.  Then the
+    // executable is rebuild *using* build-bom.
+    //
+    // Because this modifies the build, and all tests run in parallel, this first
+    // copies the build tree to a new location and then operates on that new
+    // location; this should still be faster than doing a full build in a new
+    // location.
+
+    let ZlibBld { ref tdir, ref tgt_path, .. } = *ZLIB_BLD;
+    let my_tdir = tempdir()?;
+
+    {
+        let my_tgt_path = my_tdir.path();
+        let mut cpopts = CopyOptions::new();
+        cpopts.content_only = true;
+        copy(tgt_path, my_tgt_path, &cpopts)?;
+
+        // While the above copy is intended to preserve the file timestamps, it
+        // is not always guaranteed that there will be enough system time
+        // resolution to preserve the "make finished" status, so re-run the build
+        // via build-bom to ensure it is up-to-date before making any changes.
+        let cmd_opts = vec![String::from("make"),
+                            // "-C".to_string(),
+                            // String::from(my_tgt_path.to_str().unwrap())
+            ];
+        let gen_opts = BitcodeOptions { clang_path: common::user_clang_cmd(),
+                                        bcout_path: None,
+                                        suppress_automatic_debug: false,
+                                        inject_arguments: Vec::new(),
+                                        remove_arguments: Vec::new(),
+                                        verbose: vec![true, true],
+                                        strict: false,
+                                        command: cmd_opts,
+                                        any_fail: false };
+        {
+            let _push2 = pushd(my_tgt_path)?;
+            common::gen_bitcode(gen_opts.clone())?;
+        }
+
+        let mut exe_path = PathBuf::from(my_tgt_path);
+        exe_path.push("example64");
+        assert!(exe_path.exists());  // build successfully generated a static executable
```

Comment:
You can add an extra argument to the `assert!` macro that is a string describing the assertion (instead of the comment).

Reply:
Good idea... done.